### PR TITLE
Replace README.hacking with CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by mbr0wn**
The latter is more common these days, and the former was pretty
outdated.

### CODE REVIEW

**Comment by mbr0wn** [comment]
Note: There's another changeset for GREP1 to accompany this.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+## New to GNU Radio?
+
+If you're new, it is a good idea to first get acquainted with GNU Radio from a
+user perspective.  https://tutorials.gnuradio.org/ is a good place to start.
+
+## What about non-code contributions?
+
+Those are at least as important as code contributions: Emails to the mailing
+list, answers on Stack Overflow, Wiki page edits, examples... We very much
+appreciate those. However, this document is specific about contributing code.
```
Comment (argilo) [inline CONTRIBUTING.md]:
```suggestion
appreciate those. However, this document is specifically about contributing code.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+- Keep the lines below 72 characters
+- Subject line has the component prepended (e.g., `runtime:`)
+- Avoid empty git commit messages
+- The git commit message explains the change, the code only explains the current
+  state
+
+## Include Unit Tests
+
+If you have an obvious test, that might speed up the time it takes to convince
+reviewers that your code is legit.
```
Comment (argilo) [inline CONTRIBUTING.md]:
```suggestion
reviewers that your code is correct.
```

**Comment by mbr0wn** [comment]
I picked both.

**Pull request merged**